File: leetcode/c++/7.cpp
// Hi, I'm Yanzhan. For more algothmic problems, visit my Youtube Channel (Yanzhan Yang's Youtube Channel) : https://www.youtube.com/channel/UCDkz-__gl3frqLexukpG0DA?view_as=subscriber or my Twitter Account (Yanzhan Yang's Twitter) : https://twitter.com/YangYanzhan or my GitHub HomePage (Yanzhan Yang's GitHub HomePage) : https://yangyanzhan.github.io .
// For this specific algothmic problem, visit my Youtube Video :  .
// It's fascinating to solve algothmic problems, follow Yanzhan to learn more!

class Solution {
public:
    int reverse(int x) {
        long long res = 0;
        long long tmp = x;
        int sign = tmp >= 0 ? 1 : -1;
        tmp = tmp > 0 ? tmp : -tmp;
        while (tmp) {
            int digit = tmp % 10;
            tmp /= 10;
            res *= 10;
            res += digit;
        }
        res *= sign;
        if (res > numeric_limits<int32_t>::max() || res < numeric_limits<int32_t>::min()) {
            return 0;
        }
        return res;
    }
};

File: leetcode/python/archive/7.py
#!/usr/bin/env python

__author__ = "yangyanzhan"
__email__ = "yangyanzhan@gmail.com"
__url__ = "https://github.com/yangyanzhan/code-camp"
__blog__ = "https://yangyanzhan.github.io"
__youtube__ = "https://www.youtube.com/channel/UCDkz-__gl3frqLexukpG0DA?view_as=subscriber"
__twitter__ = "https://twitter.com/YangYanzhan"

class Solution:
    def reverse(self, x: int) -> int:
        x = str(x)
        heading = ""
        if x.startswith("-"):
            heading = "-"
            x = x[1:]
        x = "".join(list(reversed(x)))
        num = float(heading + x)
        if num > 2147483647 or num < -2147483648:
            return 0
        return int(heading + x)
